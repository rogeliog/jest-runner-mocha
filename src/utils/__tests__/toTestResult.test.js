const toTestResult = require('../toTestResult');

const jestTestPath = 'path/to/file';
const start = Date.UTC(2000, 0, 1, 0, 0, 0, 0);
const end = start + 1000;

const withDescribe = (title, test) =>
  Object.assign({}, test, {
    parent: { title },
  });

const passingTest = {
  duration: 1,
  title: 'This test passes[1]',
  fullTitle: () => 'This test passes[1]',
};

const passingTest2 = {
  duration: 4,
  title: 'This test also passes[2]',
  fullTitle: () => 'This test also passes[2]',
};

const failingTest = {
  duration: 4,
  err: {
    stack: 'Error stack',
    message: 'The error message',
    showDiff: true,
    expected: 1,
    actual: 2,
  },
  title: 'This test failed',
  fullTitle: () => 'This test failed',
};

it('turns a passing mocha tests to Jest test result', () => {
  expect(
    toTestResult({
      jestTestPath,
      stats: {
        start,
        end,
        passes: 1,
        pending: 0,
        failures: 0,
      },
      tests: [passingTest],
      failures: [],
    }),
  ).toMatchSnapshot();
});

it('turns a passing mocha tests to Jest test result with coverage', () => {
  expect(
    toTestResult({
      jestTestPath,
      coverage: { mocked: 'coverageData' },
      stats: {
        start,
        end,
        passes: 1,
        pending: 0,
        failures: 0,
      },
      tests: [passingTest],
      failures: [],
    }),
  ).toMatchSnapshot();
});

it('turns a failing mocha tests to Jest test result', () => {
  expect(
    toTestResult({
      jestTestPath,
      stats: {
        start,
        end,
        passes: 0,
        pending: 0,
        failures: 1,
      },
      tests: [failingTest],
      failures: [],
    }),
  ).toMatchSnapshot();
});

it('turns a whole mocha tests suite to Jest test result', () => {
  expect(
    toTestResult({
      jestTestPath,
      stats: {
        start,
        end,
        passes: 1,
        pending: 0,
        failures: 0,
      },
      tests: [
        passingTest,
        withDescribe('Some describe block', passingTest2),
        failingTest,
      ],
      failures: [],
    }),
  ).toMatchSnapshot();
});
