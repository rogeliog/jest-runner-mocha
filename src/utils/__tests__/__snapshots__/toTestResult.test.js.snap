// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`turns a failing mocha tests to Jest test result 1`] = `
Object {
  "console": null,
  "coverage": undefined,
  "failureMessage": Array [
    "
    This test failed

      [31mThe error message[0m
      [32m+ expected[0m [31m- actual[0m

      [31m-2[0m
      [32m+1[0m
      
Error stack

",
  ],
  "numFailingTests": 1,
  "numPassingTests": 0,
  "numPendingTests": 0,
  "perfStats": Object {
    "end": 946684801000,
    "start": 946684800000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 0,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testFilePath": "path/to/file",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 0.004,
      "failureMessages": Array [
        "
    This test failed

      [31mThe error message[0m
      [32m+ expected[0m [31m- actual[0m

      [31m-2[0m
      [32m+1[0m
      
Error stack

",
      ],
      "fullName": "This test failed",
      "numPassingAsserts": 1,
      "status": "failed",
      "title": "This test failed",
    },
  ],
}
`;

exports[`turns a passing mocha tests to Jest test result 1`] = `
Object {
  "console": null,
  "coverage": undefined,
  "failureMessage": null,
  "numFailingTests": 0,
  "numPassingTests": 1,
  "numPendingTests": 0,
  "perfStats": Object {
    "end": 946684801000,
    "start": 946684800000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 0,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testFilePath": "path/to/file",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 0.001,
      "failureMessages": Array [],
      "fullName": "This test passes[1]",
      "numPassingAsserts": 0,
      "status": "passed",
      "title": "This test passes[1]",
    },
  ],
}
`;

exports[`turns a passing mocha tests to Jest test result with coverage 1`] = `
Object {
  "console": null,
  "coverage": Object {
    "mocked": "coverageData",
  },
  "failureMessage": null,
  "numFailingTests": 0,
  "numPassingTests": 1,
  "numPendingTests": 0,
  "perfStats": Object {
    "end": 946684801000,
    "start": 946684800000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 0,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testFilePath": "path/to/file",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 0.001,
      "failureMessages": Array [],
      "fullName": "This test passes[1]",
      "numPassingAsserts": 0,
      "status": "passed",
      "title": "This test passes[1]",
    },
  ],
}
`;

exports[`turns a whole mocha tests suite to Jest test result 1`] = `
Object {
  "console": null,
  "coverage": undefined,
  "failureMessage": Array [
    "
    This test failed

      [31mThe error message[0m
      [32m+ expected[0m [31m- actual[0m

      [31m-2[0m
      [32m+1[0m
      
Error stack

",
  ],
  "numFailingTests": 0,
  "numPassingTests": 1,
  "numPendingTests": 0,
  "perfStats": Object {
    "end": 946684801000,
    "start": 946684800000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "unmatched": 0,
    "updated": 0,
  },
  "sourceMaps": Object {},
  "testExecError": null,
  "testFilePath": "path/to/file",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "duration": 0.001,
      "failureMessages": Array [],
      "fullName": "This test passes[1]",
      "numPassingAsserts": 0,
      "status": "passed",
      "title": "This test passes[1]",
    },
    Object {
      "ancestorTitles": Array [],
      "duration": 0.004,
      "failureMessages": Array [],
      "fullName": "This test also passes[2]",
      "numPassingAsserts": 0,
      "status": "passed",
      "title": "This test also passes[2]",
    },
    Object {
      "ancestorTitles": Array [],
      "duration": 0.004,
      "failureMessages": Array [
        "
    This test failed

      [31mThe error message[0m
      [32m+ expected[0m [31m- actual[0m

      [31m-2[0m
      [32m+1[0m
      
Error stack

",
      ],
      "fullName": "This test failed",
      "numPassingAsserts": 1,
      "status": "failed",
      "title": "This test failed",
    },
  ],
}
`;
